crates/ln-dlc-node/src/channel.rs

```diff
@@ -0,0 +1,189 @@
+        format!(
+            "user_channel_id: {}, channel_id: {channel_id}, channel_state: {:?}, counterparty: {}, funding_txid: {:?}, created_at: {}, updated_at: {}",
+            self.user_channel_id.to_string(), self.channel_state, self.counterparty, self.funding_txid, self.created_at, self.updated_at
+        )
+        .fmt(f)
+    }
+}
+
+// We create a shadow copy of the channel as the the ldk channel does not
+// live beyond channel closure. The main purpose of
+// this shadow is to track general meta data of the
+// channel relevant for reporting purposes.
+#[derive(Debug, Clone, PartialEq)]
+pub struct Channel {
+    /// The `user_channel_id` is set by 10101 at the time the `Event::HTLCIntercepted` when
+    /// we are attempting to create a JIT channel.
+    pub user_channel_id: UserChannelId,
+    /// Until the `Event::ChannelReady` we do not have a `channel_id`, which is derived from
+    /// the funding transaction. We use the `user_channel_id` as identifier over the entirety
+    /// of the channel lifecycle.
+    pub channel_id: Option<ChannelId>,
+    pub inbound: u64,
+    pub outbound: u64,
+    /// Set at the `Event::ChannelReady`
+    pub funding_txid: Option<Txid>,
+    pub channel_state: ChannelState,
+    /// The counter party of the channel.
+    pub counterparty: PublicKey,
+    pub created_at: OffsetDateTime,
+    pub updated_at: OffsetDateTime,
+}
+
+impl Channel {
+    pub fn new(inbound: u64, outbound: u64, counterparty: PublicKey) -> Self {
+        Channel {
+            user_channel_id: UserChannelId::new(),
+            channel_state: ChannelState::Pending,
+            inbound,
+            outbound,
+            counterparty,
+            created_at: OffsetDateTime::now_utc(),
+            updated_at: OffsetDateTime::now_utc(),
+            channel_id: None,
+            funding_txid: None,
+        }
+    }
+
+    pub fn is_closed(&self) -> bool {
+        matches!(
+            self.channel_state,
+            ChannelState::ForceClosedLocal | ChannelState::ForceClosedRemote | ChannelState::Closed
+        )
+    }
+
+    pub fn close_channel(channel: Channel, reason: ClosureReason) -> Channel {
+        if channel.is_closed() {
+            tracing::warn!(%channel, "Unexpected state transition. Expected channel state to be either 'Pending' or 'Open', but was '{:?}'", channel.channel_state);
+        }
+
+        let mut channel = channel;
+        channel.channel_state = reason.into();
+        channel.updated_at = OffsetDateTime::now_utc();
+        channel
+    }
+
+    pub fn update_liquidty(
+        channel: &Channel,
+        channel_details: &Option<ChannelDetails>,
+    ) -> Result<Channel> {
+        let mut channel = channel.clone();
+        match channel_details {
+            Some(channel_details) => {
+                channel.inbound = channel_details.inbound_capacity_msat / 1000;
+                channel.outbound = channel_details.outbound_capacity_msat / 1000;
```

Comment:
🤔 I would highly recommend keeping msats around until the moment we want to display something. I think that this has been biting us in the e2e tests (I fixed some of that with #951).

Now it probably doesn't matter so much in terms of reporting, but you never know long term...